docs/tutorials/google/echoes.ipynb

```diff
@@ -0,0 +1,686 @@
+   "execution_count": 5,
+   "metadata": {
+    "cellView": "form",
+    "id": "SgIbS2Vmb5RB"
+   },
+   "outputs": [],
+   "source": [
+    "# The Google Cloud Project id to use.\n",
+    "project_id = '' #@param {type:\"string\"}\n",
+    "\n",
+    "if project_id == '':\n",
+    "    import os \n",
+    "    if 'GOOGLE_CLOUD_PROJECT' not in os.environ:\n",
+    "        raise Exception(\"Please setup project_id in this cell or set the `GOOGLE_CLOUD_PROJECT` env var to your project id.\")\n",
+    "    project_id = os.environ['GOOGLE_CLOUD_PROJECT']\n",
+    "else: \n",
+    "    import os\n",
+    "    os.environ['GOOGLE_CLOUD_PROJECT'] = project_id\n",
+    "\n",
+    "def authenticate_user():\n",
+    "    \"\"\"Runs the user through the Colab OAuth process.\n",
+    "\n",
+    "    Checks for Google Application Default Credentials and runs interactive login \n",
+    "    if the notebook is executed in Colab. In case the notebook is executed in Jupyter notebook\n",
+    "    or other IPython runtimes, no interactive login is provided, it is assumed that the \n",
+    "    `GOOGLE_APPLICATION_CREDENTIALS` env var is set or `gcloud auth application-default login`\n",
+    "    was executed already.\n",
+    "\n",
+    "    For more information on using Application Default Credentials see \n",
+    "    https://cloud.google.com/docs/authentication/production\n",
+    "    \"\"\"\n",
+    "    in_colab = False\n",
+    "    try:\n",
+    "        from IPython import get_ipython\n",
+    "        in_colab = 'google.colab' in str(get_ipython())\n",
+    "    except: \n",
+    "        # Notebook is not executed within IPython. Assuming external authentication.\n",
+    "        return \n",
+    "\n",
+    "    if in_colab: \n",
+    "        from google.colab import auth      \n",
+    "        print(\"Getting OAuth2 credentials.\")\n",
+    "        print(\"Press enter after entering the verification code.\")\n",
+    "        auth.authenticate_user(clear_output=False)\n",
+    "        print(\"Authentication complete.\")\n",
+    "    else: \n",
+    "        print(\"Notebook is not executed with Colab, assuming Application Default Credentials are setup.\") \n",
+    "\n",
+    "authenticate_user()\n",
+    "\n",
+    "print(\"Successful authentication to Google Cloud.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "oCT2mI2dlClM"
+   },
+   "source": [
+    "## Creating the circuits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "mg1bGbLxlClN"
+   },
+   "source": [
+    "The function below creates a Loschmidt echo using a random circuit for $U$ on a given set of `qubits` for a given number of `cycles`. A `pause` can be optionally applied after $U$ and before $U^\\dagger$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "id": "2VRqUT20lClP"
+   },
+   "outputs": [],
+   "source": [
+    "def create_loschmidt_echo_circuit(\n",
```

Comment:
Agree it might have value. It's a little unclear to me what the general plan around this and other QCS tools is -- tutorials vs. examples vs. library tools -- but I'm happy to take this up if there's a consensus.

Reply:
SGTM. Okay to leave this as future work.